lib/logstash/filters/geoip.rb

```diff
@@ -235,4 +237,20 @@ def tag_unsuccessful_lookup(event)
     @tag_on_failure.each{|tag| event.tag(tag)}
   end
 
+  def apply_geodata(geo_data_hash, event)
+    # don't do anything more if the lookup result is nil?
```

Comment:
That's putting the cart before the horse.  It's an interesting pass-by-reference here, as the `populate_geo_data` call is what's populating the `geo_data_hash`.  In other words, it can't be tested for emptiness until after `populate_geo_data` has been called.